saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/provider/service/servlet/filter/Saml2WebSsoAuthenticationFilter.java

```diff
@@ -109,6 +110,12 @@ public Authentication attemptAuthentication(HttpServletRequest request, HttpServ
 					"No relying party registration found");
 			throw new Saml2AuthenticationException(saml2Error);
 		}
+
+		if (AbstractAuthenticationToken.class.isAssignableFrom(authentication.getClass())) {
+			Object details = this.authenticationDetailsSource.buildDetails(request);
+			((AbstractAuthenticationToken) authentication).setDetails(details);
+		}
+
```

Comment:
Could you please extract this in a private method called `setDetails` so it becomes more readable?